docs/root/start/sandboxes/double-proxy.rst

```diff
@@ -0,0 +1,170 @@
+.. _install_sandboxes_double_proxy:
+
+Double proxy (with encryption)
+==============================
+
+This sandbox demonstrates a basic "double proxy" configuration, in which a simple Flask app
+connects to a PostgreSQL database, with two Envoy proxies in between.
+
+``Envoy (front)`` -> ``Flask`` -> ``Envoy (postgres-front)`` -> ``Envoy (postgres-back)`` -> ``PostgreSQL``
+
+This type of setup is common in a service mesh where Envoy acts as a "sidecar" between individual services.
+
+It can also be useful as a way of providing access for application servers to upstream services or
+databases that may be in a different location or subnet, outside of a service mesh or sidecar-based setup.
+
+Another common use case is with Envoy configured to provide "Points of presence" at the edges of the cloud,
+and to relay requests to upstream servers and services.
+
+This example encrypts the transmission of data between the two middle proxies.
+
+This can be useful if the proxies are phsyically separated or transmit data over untrusted networks.
+
+In order to  use the sandbox you will first need to generate the necessary SSL keys and certificates.
+
+This example walks through creating a certificate authority, and using it to create a domain key and sign
+certificates for the proxyies.
+
+.. include:: _include/docker-env-setup.rst
+
+Change to the ``examples/double-proxy`` directory.
+
+Step 3: Create a certificate authority
+**************************************
+
+First create a key for the certificate authority:
+
+.. code-block:: console
+
+   $ pwd
+   envoy/examples/double-proxy
+   $ mkdir -p certs
+   $ openssl genrsa -out certs/ca.key 4096
+   Generating RSA private key, 4096 bit long modulus (2 primes)
+   ..........++++
+   ..........................................................................................................++++
+   e is 65537 (0x010001)
+
+Now use the key to generate a certificate.
+
+If you wish, You can interactively alter the fields in the certificate.
+
+For the purpose of this example, the defaults should be sufficient.
+
+.. code-block:: console
+
+   $ openssl req -x509 -new -nodes -key certs/ca.key -sha256 -days 1024 -out certs/ca.crt
+
+   You are about to be asked to enter information that will be incorporated
+   into your certificate request.
+   What you are about to enter is what is called a Distinguished Name or a DN.
+   There are quite a few fields but you can leave some blank
+   For some fields there will be a default value,
+   If you enter '.', the field will be left blank.
+   -----
+   Country Name (2 letter code) [AU]:
+   State or Province Name (full name) [Some-State]:
+   Locality Name (eg, city) []:
+   Organization Name (eg, company) [Internet Widgits Pty Ltd]:
+   Organizational Unit Name (eg, section) []:
+   Common Name (e.g. server FQDN or YOUR name) []:
+   Email Address []:
+
+
+Step 4: Create a domain key
+***************************
+
+Create a key for the example domain:
```

Comment:
im wondering whether to create 2 separate keys here.

if i was building this *and* was using my own CA, then i would probably create two keys.

iirc doing it this way is more akin to creating CSRs from an official cert authority

Reply:
IMO I would recommend just keeping it as simple as possible. This seems fine to me.